management/server/peer.go

```diff
@@ -260,7 +263,10 @@ func (am *DefaultAccountManager) UpdatePeer(ctx context.Context, accountID, user
 		return nil, err
 	}
 
-	am.updateAccountPeers(ctx, account)
+	expired, _ := peer.LoginExpired(account.Settings.PeerLoginExpiration)
+	if peerLabelUpdated || (expired && peer.LoginExpirationEnabled) {
```

Comment:
peer expiration shouldn't influence a network map decision here, since the peer expiration goroutines will take of that